 --- plugins/qt-gui/src/licqgui.cpp (original)
+++ plugins/qt-gui/src/licqgui.cpp Sun Dec  9 02:59:55 2007
@@ -45,8 +45,6 @@

 #include <qstylefactory.h>

-CLicqGui *licqQtGui;
-
 const char *LP_Usage(void)
 {
 #ifdef USE_KDE
@@ -142,10 +140,9 @@
   setenv("KDE_DEBUG", "true", 0);
 #endif

-  licqQtGui = new CLicqGui(gui_argc, gui_argv);
-
-  int nResult = licqQtGui->Run(_licqDaemon);
-  licqQtGui->Shutdown();
+  CLicqGui licqQtGui(gui_argc, gui_argv);
+
+  int nResult = licqQtGui.Run(_licqDaemon);

   gui_argc = 0;
   gui_argv = NULL;
@@ -329,21 +326,6 @@
   free(m_szExtendedIcons);
 }

-void CLicqGui::Shutdown(void)
-{
-  gLog.Info("%sShutting down gui.\n", L_ENDxSTR);
-  gLog.ModifyService(S_PLUGIN, 0);
-  if(licqMainWindow) {
-    licqMainWindow->close();
-    delete licqMainWindow;
-    licqMainWindow = 0;
-  }
-
-  delete qApp;
-  qApp = 0;
-}
-
-
 int CLicqGui::Run(CICQDaemon *_licqDaemon)
 {
   // Register with the daemon, we want to receive all signals
@@ -359,6 +341,8 @@
   setMainWidget(licqMainWindow);
   int r = exec();
   _licqDaemon->UnregisterPlugin();
+  gLog.Info("%sShutting down gui.\n", L_ENDxSTR);
+  gLog.ModifyService(S_PLUGIN, 0);

   return r;
 }
--- plugins/qt-gui/src/licqgui.h.orig	2008-01-31 21:48:02.000000000 +0200
+++ plugins/qt-gui/src/licqgui.h	2008-01-31 21:48:18.000000000 +0200
@@ -46,7 +46,6 @@
   CLicqGui(int, char **);
   ~CLicqGui(void);
   int Run(CICQDaemon *);
-  void Shutdown(void);
   virtual bool x11EventFilter(XEvent *);
 
   virtual void commitData(QSessionManager& sm);
